src/main/java/seedu/ccacommander/logic/commands/EditEnrolmentCommand.java

```diff
@@ -0,0 +1,236 @@
+        if (enrolmentToEditList.size() == 0) {
+            throw new CommandException(String.format(Messages.MESSAGE_ENROLMENT_DOES_NOT_EXIST,
+                    memberIndex.getOneBased(), eventIndex.getOneBased()));
+        }
+        assert enrolmentToEditList.size() == 1 : "There should not be duplicate enrolments";
+
+        Enrolment enrolmentToEdit = enrolmentToEditList.get(0);
+        Enrolment editedEnrolment = createEditedEnrolment(enrolmentToEdit, editEnrolmentDescriptor);
+
+        model.setEnrolment(enrolmentToEdit, editedEnrolment);
+        model.updateFilteredEnrolmentList(PREDICATE_SHOW_ALL_ENROLMENTS);
+        model.commit(String.format(MESSAGE_COMMIT, editedEnrolment.getMemberAndEventEnrolment()));
+        return new CommandResult(String.format(MESSAGE_EDIT_ENROLMENT_SUCCESS, Messages.format(editedEnrolment)));
+    }
+
+    /**
+     * Creates and returns a {@code Enrolment} with the details of {@code enrolmentToEdit}
+     * edited with {@code editEnrolmentDescriptor}.
+     */
+    private static Enrolment createEditedEnrolment(Enrolment enrolmentToEdit,
+                                                    EditEnrolmentDescriptor editEnrolmentDescriptor) {
+        assert enrolmentToEdit != null;
+
+        Name memberName = editEnrolmentDescriptor.getMemberName().orElse(enrolmentToEdit.getMemberName());
+        Name eventName = editEnrolmentDescriptor.getEventName().orElse(enrolmentToEdit.getEventName());
+        Hours updatedHours = editEnrolmentDescriptor.getHours().orElse(enrolmentToEdit.getHours());
+        Remark updatedRemark = editEnrolmentDescriptor.getRemark().orElse(enrolmentToEdit.getRemark());
+
+        return new Enrolment(memberName, eventName, updatedHours, updatedRemark);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditEnrolmentCommand)) {
+            return false;
+        }
+
+        EditEnrolmentCommand otherEditEnrolmentCommand = (EditEnrolmentCommand) other;
+        return memberIndex.equals(otherEditEnrolmentCommand.memberIndex)
+                && eventIndex.equals(otherEditEnrolmentCommand.eventIndex)
+                && editEnrolmentDescriptor.equals(otherEditEnrolmentCommand.editEnrolmentDescriptor);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("memberIndex", memberIndex)
+                .add("eventIndex", eventIndex)
+                .add("editEnrolmentDescriptor", editEnrolmentDescriptor)
+                .toString();
+    }
+
+    /**
+     * Stores the details to edit the enrolment with. Each non-empty field value will replace the
+     * corresponding field value of the event.
+     */
+    public static class EditEnrolmentDescriptor {
+        private Name memberName;
+        private Name eventName;
```

Comment:
should we leave Names related fields and methods in EditEnrolmentDescriptor out, cause i checked and it seemed they arent used in any place below. And because memberName and eventName are not supposed to be updated. But either way it wouldnt really affect 👍

Reply:
I think they will be required for editing memberName of a member, or eventName of an event, as these fields will then have to be edited. For example, editing a member's Name will result in the editing of all their Enrolments.